zenodb_test.go

```diff
@@ -100,11 +105,24 @@ func doTestCluster(t *testing.T, numPartitions int, partitionBy []string) {
 						leader.RegisterQueryHandler(partition, func(ctx context.Context, sqlString string, isSubQuery bool, subQueryResults [][]interface{}, unflat bool, onFields core.OnFields, onRow core.OnRow, onFlatRow core.OnFlatRow) error {
 							// Re-register when finished
 							defer register()
-							return query(ctx, sqlString, isSubQuery, subQueryResults, unflat, onFields, onRow, onFlatRow)
+							return query(ctx, sqlString, isSubQuery, subQueryResults, unflat, onFields, func(key bytemap.ByteMap, vals core.Vals) (bool, error) {
+								copyOfKey := make(bytemap.ByteMap, len(key))
+								copy(copyOfKey, key)
+								copyOfVals := make(core.Vals, 0, len(vals))
+								for _, val := range vals {
+									copyOfVal := make(encoding.Sequence, len(val))
+									copy(copyOfVal, val)
+									copyOfVals = append(copyOfVals, copyOfVal)
+								}
+								return onRow(copyOfKey, copyOfVals)
+							}, onFlatRow)
 						})
 					}
 
-					register()
+					// Continously handle queries
+					for j := 0; j < clusterQueryConcurrency; j++ {
```

Comment:
This was the fix for the test. The unit test mimics the cluster functionality but doesn't actually use the real gRPC-based protocol. In the cluster, query processors need to register multiple query handlers to handle parallel queries, so this accomplishes that.